graphql/subcourse/mutations.ts

```diff
@@ -250,20 +250,20 @@ export class MutateSubcourseResolver {
     }
 
     @Mutation((returns) => Boolean)
-    @AuthorizedDeferred(Role.ADMIN, Role.SUBCOURSE_PARTICIPANT)
+    @AuthorizedDeferred(Role.ADMIN, Role.SUBCOURSE_PARTICIPANT, Role.OWNER)
     async subcourseLeave(
         @Ctx() context: GraphQLContext,
         @Arg('subcourseId') subcourseId: number,
         @Arg('pupilId', { nullable: true }) pupilId?: number
     ): Promise<boolean> {
-        const { user } = context;
-        const pupil = await getSessionPupil(context, pupilId);
-        const pupilUser = userForPupil(pupil);
         const subcourse = await getSubcourse(subcourseId);
         await hasAccess(context, 'Subcourse', subcourse);
 
+        const pupil = await getPupil(pupilId || context.user.pupilId);
```

Comment:
Participants could now throw out other users. pupilId may only be passed in by Owner and Admin.

Reply:
Very good point. I've pushed an update.